specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingSubscription.json

```diff
@@ -400,244 +749,605 @@
+        },
+        "value": {
+          "description": "The list of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BillingSubscription"
+          },
+          "readOnly": true,
+          "x-ms-identifiers": [
+            "id"
+          ]
+        }
+      }
+    },
+    "BillingSubscriptionPatch": {
+      "description": "The billing properties of a subscription.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./types.json#/definitions/ArmPatchResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "The properties of a(n) BillingSubscription",
+          "allOf": [
+            {
+              "$ref": "#/definitions/BillingSubscriptionProperties"
+            }
+          ],
           "x-ms-client-flatten": true
         }
       },
-      "type": "object"
+      "x-ms-azure-resource": true
     },
     "BillingSubscriptionProperties": {
       "description": "The billing properties of a subscription.",
+      "type": "object",
       "properties": {
-        "billingProfileDisplayName": {
-          "description": "The name of the billing profile to which the subscription is billed.",
+        "autoRenew": {
+          "description": "Indicates whether auto renewal is turned on or off for a product.",
+          "enum": [
+            "Off",
+            "On"
+          ],
+          "type": "string",
           "readOnly": true,
+          "x-ms-enum": {
+            "name": "AutoRenew",
+            "modelAsString": true
+          }
+        },
+        "beneficiary": {
+          "description": "Details of the beneficiary.",
+          "allOf": [
+            {
+              "$ref": "#/definitions/Beneficiary"
+            }
+          ]
+        },
+        "billingFrequency": {
+          "description": "The billing frequency in ISO8601 format of product in the subscription. Example: P1M, P3M, P1Y",
           "type": "string"
         },
         "billingProfileId": {
-          "description": "The ID of the billing profile to which the subscription is billed.",
-          "readOnly": true,
+          "description": "The fully qualified ID that uniquely identifies a billing profile.",
           "type": "string"
         },
-        "costCenter": {
-          "description": "The cost center applied to the subscription.",
-          "type": "string"
+        "billingProfileDisplayName": {
+          "description": "The name of the billing profile for the invoice section or customer.",
+          "type": "string",
+          "readOnly": true
         },
-        "customerDisplayName": {
-          "description": "The name of the customer for whom the subscription was created. The field is applicable only for Microsoft Partner Agreement billing account.",
-          "readOnly": true,
+        "billingProfileName": {
+          "description": "The ID that uniquely identifies a billing profile.",
```

Comment:
Why do we need this? This is already included in billingProfileId. I don't believe other ARM resources do this when referencing other resources, do they? Presumably they all expect a client to grab the resource name from the id uri...

Reply:
There is precedent in BillingRP APIs for having both the fully-qualified id and the short id.